Document template vars (#286)
cylc/cylc-doc: src/user-guide/writing-workflows/jinja2.rst

.. _jinja2-template-variables:

Jinja2 Default Values And Template Variables
--------------------------------------------

You can provide template variables to Cylc in 3 ways:

- Using the ``--set-file`` option.
- Using the ``-s`` option.

> - Using a plugin, such as :ref:`Cylc Rose`.

Reviewer: Would be nice if these could link to the sections below. But not essential

Author: ```suggestion
- :ref:`Using a plugin`, such as :ref:`Cylc Rose`.
```

.. note::

>    If the same variable is set by more than one method the last source in the

Reviewer: ```suggestion
   If the same variable is set by more than one method, the last source in the
```

   be overridden.

Using a plugin
^^^^^^^^^^^^^^

Template plugins such as :ref:`Cylc Rose` should provide a set of template
variables which can be provided to Cylc. For example, using Cylc Rose you

> add a ``rose-suite.conf`` file which contains a ``[template variables]``
> section which the plugin makes available to Cylc:
> 
> For example, with Cylc Rose installed:

Reviewer: ```suggestion
add a ``rose-suite.conf`` file containing a ``[template variables]``
section which the plugin makes available to Cylc:
```

> .. code-block:: ini
> 
>    # rose-suite.conf

Reviewer: ```suggestion
.. code-block:: ini
   :caption: rose-suite.conf
   
```

   [template variables]

>    ICP = 1068

Author: ```suggestion
ICP=1067
```

> .. code-block:: cylc

Reviewer: ```suggestion
.. code-block:: cylc
   :caption: flow.cylc
```

   #!jinja2
   [scheduler]
      allow implicit tasks = True
   [scheduling]
      initial cycle point = {{ICP}}
      [[dependencies]]
         P1Y = Task1

.. code-block:: shell

   > cylc config . -i "[scheduling]initial cycle point"
   1068

> 

Reviewer 2: Not sure what I'm doing wrong here @wxtim, hopefully you can spot my mistake...I've tried out this example but hitting an error...I'm on your branches for https://github.com/cylc/cylc-flow/pull/4360 and the cylc-rose sibling, having pip installed them both and I'm getting.... 
![image](https://user-images.githubusercontent.com/37735232/129911100-667164f7-8f8a-401f-b121-50ae395f7a13.png)

Author: I don't think that rose supports spaces in `key = value`. Try `ICP=1068`

Reviewer 2: I tried without the spaces :( No luck, same error. My suspicion is it is my environment, I will try on a fresh one and see if that helps.

Author: :(
